### ISSUE TITLE
Improve requirements structure

### ISSUE THREAD

**Issue by spflueger** [ComPWA/tensorwaves#28]
For me the general way the requirements are currently setup is not ideal.
https://stackoverflow.com/questions/17803829/how-to-customize-a-requirements-txt-for-multiple-environments
I would follow this. So having requirements files for each environment
- ci/test
- prod
- dev

and reuse common dependencies across them. Then if you are a dev, you simply run the dev req and you are all setup and so on.

**Comment by redeboer** [comment]
Probably better to solve with through either a [setup.cfg](https://docs.python.org/3/distutils/configfile.html) file or pyproject.toml (see [PEP 518](https://www.python.org/dev/peps/pep-0518/)).

**Comment by redeboer** [comment]
We now again have the problem in #88. The issue is that there isn't one source of truth for the run dependencies. That is, `setup.py` for `setuptools` (used when deploying to PyPI) and `requirements.txt` (used in 'developer mode', see e.g. [`environment.yml`](https://github.com/ComPWA/tensorwaves/blob/6646b5377379b763233fcd9c72e34f5da147663a/environment.yml) for Conda).

**Comment by spflueger** [comment]
I found this as a pretty good resource https://snarky.ca/what-the-heck-is-pyproject-toml/

Btw this issue is related to
ComPWA/expertsystem#68
ComPWA/expertsystem#70

**Comment by redeboer** [comment]
> Btw this issue is related to ...

Yes good point, the problem there is that there isn't one source of truth for the version number of the package
